File: backend/lite/exp.py
def isVariable(exp): return isinstance(exp,str)
def isSelfEvaluating(exp): return not isinstance(exp,list)

def isQuotation(exp): 
  assert isinstance(exp,list)
  assert len(exp) > 0
  return exp[0] == "quote"

def textOfQuotation(exp): 
  assert len(exp) > 1
  return exp[1] 

def getOperator(exp): 
  assert isinstance(exp,list)
  assert len(exp) > 0
  return exp[0]

def getOperands(exp): return exp[1:]

File: backend/jventure/exp.jl
# isVariable must precede isSelfEvaluating

isVariable(sym::Symbol) = true
isVariable(x::VentureValue) = false

isSelfEvaluating(xs::Array{VentureValue}) = false
isSelfEvaluating(x::VentureValue) = true

isQuotation(xs::Array{VentureValue}) = (xs[1] == :quote)
textOfQuotation(xs::Array{VentureValue}) = xs[2]

getOperator(xs::Array{VentureValue}) = xs[1]
getOperands(xs::Array{VentureValue}) = xs[2:]